src/proto_tcp.c

```diff
@@ -141,6 +141,41 @@ struct protocol proto_tcpv6 = {
 
 INITCALL1(STG_REGISTER, protocol_register, &proto_tcpv6);
 
+
+struct protocol proto_mptcpv4;
+struct protocol proto_mptcpv6;
+
+static void mptcp_protocol_register(struct protocol *proto)
+{
+	proto->sock_prot = IPPROTO_MPTCP;
+	protocol_register(proto);
+}
+
+static void mptcpv4_protocol_register(struct protocol *proto)
+{
+	proto_mptcpv4 = proto_tcpv4;
+	strcpy(proto_mptcpv4.name, "mptcpv4");
+	printf("proto_mptcpv4.name: %s\n", proto_mptcpv4.name);
+	printf("proto_mptcpv4.sock_type: %d\n", proto_mptcpv4.sock_type);
+	printf("proto_mptcpv4.sock_prot: %d\n", proto_mptcpv4.sock_prot);
+	printf("proto_mptcpv4.fam->name: %s\n\n\n", proto_mptcpv4.fam->name);
+	printf("proto_mptcpv4.nb_receivers: %d\n", proto_mptcpv4.nb_receivers);
+	printf("proto_mptcpv4.receivers: %p\n\n\n", &proto_mptcpv4.receivers);
+	mptcp_protocol_register(&proto_mptcpv4);
+}
+
+static void mptcpv6_protocol_register(struct protocol *proto)
+{
+	proto_mptcpv6 = proto_tcpv6;
+	strcpy(proto_mptcpv6.name, "mptcpv6");
+	mptcp_protocol_register(&proto_mptcpv6);
+}
+
+
+// INITCALL1(STG_REGISTER, mptcpv4_protocol_register, &proto_mptcpv4);
+// INITCALL1(STG_REGISTER, mptcpv6_protocol_register, &proto_mptcpv6);
```

Comment:
detail: here, you can pass `NULL` instead of `&proto_mptcpv4` if you don't use `proto` above.

(but I guess you want this to be not commented)